src/views/HomeRedesign/HomeSection.tsx

```diff
@@ -0,0 +1,68 @@
+import { Heading } from "@chakra-ui/react";
+import { Fragment, FunctionComponent, useMemo } from "react";
+import { CatalogPackage } from "../../api/package/packages";
+import { PackageList } from "../../components/PackageList";
+import { useCatalog } from "../../contexts/Catalog";
+
+export interface HomeSectionProps {
+  name: string;
+  showLastUpdated?: number;
+  showPackages?: string[];
+}
+
+export const HomeSection: FunctionComponent<HomeSectionProps> = ({
+  name,
+  showLastUpdated,
+  showPackages,
+}) => {
+  const { data, loading, error } = useCatalog();
+
+  const results = useMemo(() => {
+    if (loading || error || !data?.packages) return [];
+
+    let packages = data.packages;
+
+    if (showLastUpdated) {
+      // sort packages by date
+      packages = [...packages]
+        .sort((p1, p2) => {
+          const d1 = new Date(p1.metadata.date);
+          const d2 = new Date(p2.metadata.date);
+          if (d1 === d2) {
+            return 0;
+          }
+          return d1 < d2 ? 1 : -1;
+        })
+        .slice(0, showLastUpdated);
+    } else if (showPackages) {
+      // find the specific packages from the catalog and
+      // return them in the same order
+      const filteredMap = [...packages].reduce(
+        (accum: { [key: string]: CatalogPackage }, p) => {
+          if (showPackages.includes(p.name)) {
+            return {
+              [p.name]: p,
+              ...accum,
+            };
+          } else {
+            return accum;
+          }
+        },
+        {}
+      );
+      packages = showPackages.map((p) => filteredMap[p]);
+    }
+
+    return packages;
+  }, [data?.packages, error, loading, showLastUpdated, showPackages]);
```

Comment:
This is my own opinion, but I believe this memo should only return packages if either `showLastUpdated` or `showPackages` is defined - as otherwise it means we received an invalid config. As such, we should default `packages` to undefined rather than `data.packages`. We would then want to have our component conditionally return null if `results` is not defined like the following:

```tsx
if (!results) {
  console.warn("Helpful message about your invalid config");
  return null;
}

return (
  ...
);
```

Reply:
Good point, if the code doesn't break/return it might try rendering all of the packages in the catalog which would be pretty bad...

Reply:
I didn't even think of that edge-case, but yeah that'd cause a pretty noticeable stutter 😅